### PR TITLE
Improve tests for Array.ConstrainedCopy

### PR DESCRIPTION
**Pull request by hughbe**
- And CopyTo

/cc @stephentoub

### CODE REVIEW

File: src/System.Runtime/tests/System/ArrayTests.cs [hunk 1]
```diff
@@ -754,52 +754,8 @@ public static void ConstrainedCopy_Invalid()
...
-
-            int[] intArray2 = new int[] { 0x12345678, 0x22334455, 0x778899aa, 0x55443322, 0x33445566 };
-            yield return new object[] { intArray2, 2, intArray2, 3, 2, new int[] { 0x12345678, 0x22334455, 0x778899aa, 0x778899aa, 0x55443322 } };
-
-            yield return new object[] { new string[] { "Red", "Green", null, "Blue" }, 0, new string[] { "X", "X", "X", "X" }, 0, 4, new string[] { "Red", "Green", null, "Blue" } };
-
-            string[] stringArray = new string[] { "Red", "Green", null, "Blue" };
-            yield return new object[] { stringArray, 1, stringArray, 2, 2, new string[] { "Red", "Green", "Green", null } };
-
-            // Value type array to reference type array
```
Comment (hughbe) [inline src/System.Runtime/tests/System/ArrayTests.cs]: L772-786 moved here:
https://github.com/dotnet/corefx/pull/11603/commits/91a12384ec6b64c179d50ea29448e96aa9095e20#diff-5b0c2a1222da27a6933bd14cfdfe7f01R974

File: src/System.Runtime/tests/System/ArrayTests.cs [hunk 2]
```diff
@@ -754,52 +754,8 @@ public static void ConstrainedCopy_Invalid()
             Assert.Throws<ArgumentOutOfRangeException>("length", () => Array.ConstrainedCopy(new string[10], 0, new string[10], 0, -1)); // Length < 0
         }
 
-        public static IEnumerable<object[]> Copy_TestData()
+        public static IEnumerable<object[]> Copy_Array_Reliable_TestData()
         {
-            yield return new object[] { new int[] { 0x12345678, 0x22334455, 0x778899aa }, 0, new int[3], 0, 3, new int[] { 0x12345678, 0x22334455, 0x778899aa } };
```
Comment (hughbe) [inline src/System.Runtime/tests/System/ArrayTests.cs]: L759-770 moved here: https://github.com/dotnet/corefx/pull/11603/commits/91a12384ec6b64c179d50ea29448e96aa9095e20#diff-5b0c2a1222da27a6933bd14cfdfe7f01R848

File: src/System.Runtime/tests/System/ArrayTests.cs [hunk 3]
```diff
@@ -754,52 +754,8 @@ public static void ConstrainedCopy_Invalid()
...
-            yield return new object[] { new IEquatable<int>[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 }, 2, new int[] { 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc }, 5, 3, new int[] { 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 2, 3, 4, 0xcc, 0xcc } };
-            yield return new object[] { new IEquatable<int>[] { 0, new NotInt32(), 2, 3, 4, new NotInt32(), 6, 7, 8, 9 }, 2, new int[] { 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc }, 5, 3, new int[] { 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 2, 3, 4, 0xcc, 0xcc } };
-
-            yield return new object[] { new object[] { 0, 1, 2, 3, null, 5, 6, 7, 8, 9 }, 2, new int?[] { 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 0xcc }, 5, 3, new int?[] { 0xcc, 0xcc, 0xcc, 0xcc, 0xcc, 2, 3, null, 0xcc, 0xcc } };
-
-            // Struct[] -> object[]
-            NonGenericStruct[] structArray1 = CreateStructArray();
-            yield return new object[] { structArray1, 0, new object[5], 0, 5, structArray1.Select(g => (object)g).ToArray() };
-
-            // Struct[] -> Struct[]
```
Comment (hughbe) [inline src/System.Runtime/tests/System/ArrayTests.cs]: L788-802 moved here: https://github.com/dotnet/corefx/pull/11603/commits/91a12384ec6b64c179d50ea29448e96aa9095e20#diff-5b0c2a1222da27a6933bd14cfdfe7f01R862

File: src/System.Runtime/tests/System/ArrayTests.cs [hunk 4]
```diff
@@ -815,27 +771,14 @@ public static IEnumerable<object[]> Copy_TestData()
             yield return new object[] { intRank2Array, 0, new int[3, 2], 0, 6, new int[,] { { 1, 2 }, { 3, 4 }, { 5, 6 } } };
             yield return new object[] { intRank2Array, 1, new int[2, 3], 2, 3, new int[,] { { 0, 0, 2 }, { 3, 4, 0 } } };
 
-            // int[,] -> long[,]
```
Comment (hughbe) [inline src/System.Runtime/tests/System/ArrayTests.cs]: L818-831 move here: https://github.com/dotnet/corefx/pull/11603/commits/91a12384ec6b64c179d50ea29448e96aa9095e20#diff-5b0c2a1222da27a6933bd14cfdfe7f01R993

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**